src/mpid/ch4/netmod/ofi/ofi_recv.h

```diff
@@ -289,6 +289,58 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_OFI_do_irecv(void *buf,
 }
 
 
+#ifdef MPL_HAVE_GPU
+static inline int MPIDI_NM_do_gpu_recv(void *buf,
```

Comment:
This function is very similar to `MPIDI_NM_do_gpu_irecv`. Both of them are doing pointer check, staging recv buffer and calling internode CPU transfer.

In future, if `imrecv` is supported, we will add another `MPIDI_NM_do_gpu_imrecv` with lots of duplication.

Is it possible to have one and reuse it for `NM_recv`, `NM_irecv` and `NM_imrecv` so on?

Reply:
I think it is the same case for `send`, is it possible to have only one `MPIDI_NM_do_gpu_isend`, which can be reused for `send`, `isend`, 'ssend'?

Reply:
Yes it should be possible to reuse these but adding additional flags as `NM_recv` and `NM_irecv` have different `MPIDIG_xxx` fallbacks.

Reply:
On second thought I don't think it should be too problematic to have a little bit of duplication as these function will go away eventually when libfabric adds native support for GPU memory. But if this is important to you we can change them.

Reply:
That's fine, we can keep it for now. I think `isend` and `irecv` are most important functions to support in the beginning.

Reply:
I have changed the `NM_do_gpu_xxx` function names to a more generic `NM_do_xxx`. We usually use the `do_xxx` in the name to indicate functions that are used locally.